File: src/main/java/com/ject/studytrip/global/exception/response/ErrorResponse.java
package com.ject.studytrip.global.exception.response;

public record ErrorResponse(String error, String message, Object values) {

    public static ErrorResponse of(String error, String message, Object values) {
        return new ErrorResponse(error, message, values);
    }

    public static ErrorResponse of(String error, String message) {
        return new ErrorResponse(error, message, null);
    }
}

File: src/main/kotlin/com/ject/studytrip/global/exception/response/ErrorResponse.kt
package com.ject.studytrip.global.exception.response

data class ErrorResponse(
    val error: String,
    val message: String,
    val values: Any?,
) {
    companion object {
        fun of(
            error: String,
            message: String,
            values: Any?,
        ): ErrorResponse = ErrorResponse(error, message, values)

        fun of(
            error: String,
            message: String,
        ): ErrorResponse = ErrorResponse(error, message, null)
    }
}
